src/librustdoc/html/render/print_item.rs

```diff
@@ -130,6 +130,23 @@ pub(super) fn print_item(cx: &Context<'_>, item: &clean::Item, buf: &mut Buffer)
     }
 }
 
+/// For large structs, enums, unions, etc, determine whether to hide their fields
+fn should_hide_fields(n_fields: usize) -> bool {
+    // todo: figure out what this should be
+    n_fields > 5
+}
+
+fn toggle_open(w: &mut Buffer, text: &str)
+{
+    write!(w, "<div class=\"docblock type-contents-toggle\" data-toggle-text=\"{}\">", text);
+}
+
+fn toggle_close(w: &mut Buffer)
+{
+    w.write_str("</div>");
+}
```

Comment:
nit: it might be nice to make this an RAII type that closes the toggle on drop, so you can't forget to close it. But maybe that would be confusing; I'll leave it up to you.

Reply:
Yeah, I think that would be confusing, and it would be tricky because the `toggle_open` is inside an `if` block, so I think it would be dropped too soon. Maybe a better approach would be to use a `with_toggle` function that takes a closure? It would look like this:

```rust
with_toggle(|| /* do stuff that should be behind a toggle */ );
```

However, that wouldn't work with the conditional toggling determined by the `toggle` variable. Instead we could do something like

```rust
maybe_with_toggle(toggle, || /* do stuff */);
```

where a toggle is only added if `toggle` is true?

Reply:
@jyn514 No, I want the toggle to close at a _very specific_ point.

I thought about adding a closure: I think a problem is that the closure would get _really weird_ for the `trait` stuff. It's cleaner this way, I _started out_ with the other design.

Reply:
Maybe at least add an `#[inline]` attribute considering it's a just one line?

Reply:
I think this is fine, rustdoc has lots of other performance issues and I wouldn't expect this to make a difference. I didn't realized you'd tried this first, thanks for testing it out :)